shell/shared/grain.js

```diff
@@ -511,6 +511,62 @@ if (Meteor.isClient) {
             currentGrainId + event.data.setPath);
       } else if (event.data.setTitle) {
         Session.set("grainFrameTitle", event.data.setTitle);
+      } else if (event.data.renderTemplate) {
+        // Request creation of a single-use template with a privileged API token.
+        // Why?  Apps should not be able to obtain capabilities-as-keys to
+        // themselves directly, because those can be leaked through an
+        // arbitrary bit stream or covert channel.  However, apps often need a
+        // way to provide instructions to users to copy/paste with some
+        // privileged token contained within.  By providing this templating in
+        // the platform, we can ensure that the token is only visible to the
+        // shell's origin.
+        var call = event.data.renderTemplate;
+        check(call, Object);
+        var rpcId = call.rpcId;
+        check(rpcId, String);
+        var template = call.template;
+        check(template, String);
+        var assignment = undefined;
+        if (call.roleId) {
```

Comment:
Oops, sorry. I meant to ask "Is there any reason **not** to allow an arbitrary Grain.ViewSharingLink.RoleAssignment here?"

The allAccess role carries all permissions. This means that whoever uses the token will get exactly the permissions of the user who created the token. As discussed in https://github.com/sandstorm-io/sandstorm/pull/443, this is commonly the right thing.